airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/mysql/MySqlDebeziumStateUtil.java

```diff
@@ -0,0 +1,141 @@
+    properties.setProperty("snapshot.mode", "schema_only_recovery");
+    final AirbyteFileOffsetBackingStore offsetManager = AirbyteFileOffsetBackingStore.initializeState(
+        constructBinlogOffset(database, database.getSourceConfig().get(JdbcUtils.DATABASE_KEY).asText()),
+        Optional.empty());
+    final AirbyteSchemaHistoryStorage schemaHistoryStorage = AirbyteSchemaHistoryStorage.initializeDBHistory(Optional.empty());
+    final LinkedBlockingQueue<ChangeEvent<String, String>> queue = new LinkedBlockingQueue<>();
+    try (final DebeziumRecordPublisher publisher = new DebeziumRecordPublisher(properties, database.getSourceConfig(), catalog, offsetManager,
+        Optional.of(schemaHistoryStorage))) {
+      publisher.start(queue);
+      while (!publisher.hasClosed()) {
+        final ChangeEvent<String, String> event = queue.poll(10, TimeUnit.SECONDS);
+        if (event == null) {
+          continue;
+        }
+        LOGGER.info("A record is returned, closing the engine since the state is constructed");
+        publisher.close();
+        break;
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    final Map<String, String> offset = offsetManager.read();
+    final String dbHistory = schemaHistoryStorage.read();
+
+    assert !offset.isEmpty();
+    assert Objects.nonNull(dbHistory);
+
+    final Map<String, Object> state = new HashMap<>();
+    state.put(MYSQL_CDC_OFFSET, offset);
+    state.put(MYSQL_DB_HISTORY, dbHistory);
+
+    final JsonNode asJson = Jsons.jsonNode(state);
+    LOGGER.info("Initial Debezium state constructed: {}", asJson);
+
+    return asJson;
+  }
+
+  /**
+   * Method to construct initial Debezium state which can be passed onto Debezium engine to make it
+   * process binlogs from a specific file and position and skip snapshot phase
+   */
+  private JsonNode constructBinlogOffset(final JdbcDatabase database, final String dbName) {
+    return format(getStateAttributesFromDB(database), dbName, Instant.now());
+  }
+
+  @VisibleForTesting
+  public JsonNode format(final MysqlDebeziumStateAttributes attributes, final String dbName, final Instant time) {
+    final String key = "[\"" + dbName + "\",{\"server\":\"" + dbName + "\"}]";
+    final String gtidSet = attributes.gtidSet().isPresent() ? ",\"gtids\":\"" + attributes.gtidSet().get() + "\"" : "";
+    final String value =
+        "{\"transaction_id\":null,\"ts_sec\":" + time.getEpochSecond() + ",\"file\":\"" + attributes.binlogFilename() + "\",\"pos\":"
+            + attributes.binlogPosition()
+            + gtidSet + "}";
+
+    final Map<String, String> result = new HashMap<>();
+    result.put(key, value);
+
+    final JsonNode jsonNode = Jsons.jsonNode(result);
+    LOGGER.info("Initial Debezium state constructed: {}", jsonNode);
+
+    return jsonNode;
+  }
+
+  private MysqlDebeziumStateAttributes getStateAttributesFromDB(final JdbcDatabase database) {
```

Comment:
tiny nit : let's make this method public. I'm going to need it in the initial snapshotting code, so that we can attach the binlog_file + position for each record emitted in the initial pk sync